src/Components/test/testassets/TestServer/Startup.cs

```diff
@@ -47,8 +49,11 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 // we're not relying on any extra magic inside UseServerSideBlazor, since it's
                 // important that people can set up these bits of middleware manually (e.g., to
                 // swap in UseAzureSignalR instead of UseSignalR).
-                subdirApp.UseSignalR(route => route.MapHub<ComponentsHub>(ComponentsHub.DefaultPath));
-                subdirApp.UseBlazor<BasicTestApp.Startup>();
+                subdirApp.UseRouting(routes => routes.MapComponentHub<Index>(ComponentHub.DefaultPath, selector: "root"));
+
+                subdirApp.MapWhen(
+                    ctx => ctx.Features.Get<IEndpointFeature>()?.Endpoint == null,
+                    blazorBuilder => blazorBuilder.UseBlazor<BasicTestApp.Startup>());
```

Comment:
> ctx => ctx.Features.Get<IEndpointFeature>()?.Endpoint == null

How is this condition ever true? Isn't it always false because of line 29?

Reply:
If routing doesn't match, and endpoint doesn't get assigned. No sure what you mean because the diff is not clear. But the E2E tests pass